### ISSUE TITLE
Support for escaping parameters

### ISSUE THREAD

**Issue by jamescun** [julienschmidt/httprouter#196]
I'm attempting to implement a REST API following the Google API Design guidelines, and have hit a snag while trying to implement [Custom Methods](https://cloud.google.com/apis/design/custom_methods).

Creating a route like `/foo:customVerb` will result in `:customVerb` becoming a parameter rather than part of the path. It would be helpful to be able to escape the parameter colon (perhaps like `/foo\:customVerb`).

**Comment by david-l-riley** [comment]
Agreed, it would be easy enough to make a PR for this.  A simple check for a backslash escape when looking for colons and asterisks would be enough; better yet, also validate by making sure the colon or star is preceded by a path delimiter, because variables don't generally happen in the middle of words (though that may be desirable behavior).  Would it help to make a PR?

**Comment by bharat42** [comment]
Liked gin gonic but have to choose another one for my project for the same reason.

**Comment by leefernandes** [comment]
Or allowing the named parameter prefix character/s to be specified at compile time, allowing alternatives to `:`
